ts/features/fims/components/FimsWebView.tsx

```diff
@@ -77,67 +77,73 @@ const FimsWebView = (props: Props) => {
-
-    const locale = getRemoteLocale();
-    const message = maybeData.right;
-    switch (message.type) {
-      case "CLOSE_MODAL":
-        props.onWebviewClose();
-        return;
-      case "START_LOAD":
-        setLoading(true);
-        return;
-      case "END_LOAD":
-        setLoading(false);
-        return;
-      case "SHOW_SUCCESS":
-        showModal(
-          <SuccessContent
-            text={pipe(
-              message[locale],
-              O.fromNullable,
-              O.getOrElse(() => message.en)
-            )}
-            close={() => {
-              hideModal();
-              props.onWebviewClose();
-            }}
-          />
-        );
-        return;
-      case "SHOW_ERROR":
-        showModal(
-          <ErrorContent
-            text={pipe(
-              message[locale],
-              O.fromNullable,
-              O.getOrElse(() => message.en)
-            )}
-            close={hideModal}
-          />
-        );
-        return;
-      case "SHOW_ALERT":
-        const value = pipe(
-          message[locale],
-          O.fromNullable,
-          O.getOrElse(() => message.en)
-        );
-        Alert.alert(value.title, value.description);
-        return;
-      default:
-        return;
-    }
+    pipe(
+      JSON.parse(event.nativeEvent.data),
+      WebviewMessage.decode,
+      E.fold(
+        () => {
+          showToast(I18n.t("webView.error.convertMessage"));
+        },
+        message => {
+          const locale = getRemoteLocale();
+
+          switch (message.type) {
+            case "CLOSE_MODAL":
+              onWebviewClose();
+              break;
+            case "START_LOAD":
+              setLoading(true);
+              break;
+            case "END_LOAD":
+              setLoading(false);
+              break;
+            case "SHOW_SUCCESS":
+              showModal(
+                <SuccessContent
+                  text={pipe(
+                    message[locale],
+                    O.fromNullable,
+                    O.getOrElse(() => message.en)
+                  )}
+                  close={() => {
+                    hideModal();
+                    onWebviewClose();
+                  }}
+                />
+              );
+              break;
+            case "SHOW_ERROR":
+              showModal(
+                <ErrorContent
+                  text={pipe(
+                    message[locale],
+                    O.fromNullable,
+                    O.getOrElse(() => message.en)
+                  )}
+                  close={hideModal}
+                />
+              );
+              break;
+            case "SHOW_ALERT":
+              const value = pipe(
+                message[locale],
+                O.fromNullable,
+                O.getOrElse(() => message.en)
+              );
+              Alert.alert(value.title, value.description);
+              break;
+            case "SET_TITLE":
+              const title = pipe(
```

Comment:
The `pipe` sequence is repeated throughout multiple cases. What if we define a (cached) function that encapsulates its logic? Like
```
const pickAName = useCallback(
  (message: ProperType, locale: string) => 
    pipe(
      message[locale], 
      O.fromNullable, 
      O.getOrElse(() => message.en
    )
);    
```

Reply:
Changed https://github.com/pagopa/io-app/pull/4744/commits/cbd660ee14e590161256da592e44222fbf027dba 
Through this amendment, I omitted the `SHOW_SUCCESS` and `SHOW_ERROR` cases because they are going to be removed in this [PR](https://github.com/pagopa/io-app/pull/4744)